pkg/systemd/ssh-list-host-keys.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/sh
+
+set -eu
+port="22"
+host="localhost"
+
+parse_addr() {
+    if [ -n "$1" ]; then
+        count=$(echo "$1" | grep -o ":" | wc -l)
+        if [ "$count" -ne "0" ]; then
+            p_spot=$(($count + 1))
+            p=$(echo "$1" | cut -d ":" "-f$p_spot")
+            h=$(echo "$1" | cut -d ":" "-f-$count")
+        else
+            p=$1
+            h="localhost"
+        fi
+
+        if [ "$p" -eq "$p" ] 2>/dev/null; then
+            port="$p"
+            host="$h"
+            if [ "$host" == "[::]" ]; then
+                host="::"
+            fi
+        fi
+    fi
+}
+
+# Try to find where sshd might be listening
+
+# Check sshd_config, only works for root
+$(sshd -T | grep "listenaddress " | cut -d' ' -f2-) | while IFS='\n' read line; do
+    parse_addr "$line"
+done
+
+# Check with systemd
+SYSTEMD_TARGET="/etc/systemd/system/sockets.target.wants/sshd.socket"
+systemd_address=$(cat "$SYSTEMD_TARGET" | grep "ListenStream=" | cut -d"=" -f2-)
+parse_addr "$systemd_address"
+
+for algo in $(ssh -Q key-plain); do
```

Comment:
Why do you need to go through each algorithm individually? When I use the following command, it lists all keys:

```
$ ssh-keyscan -p 22 localhost
```